src/hal/HAL.h

```diff
@@ -11,14 +11,17 @@
 #include "Motor.h"
 #include "SwitchPoint.h"
 #include "TrafficLight.h"
+#include "Observable.h"
 
 namespace hal {
 
-	class HAL {
+	class HAL : public Observable{
 	public:
 		HAL();
 		virtual ~HAL();
 
+		string getSignal();
```

Comment:
hier kannst du auch schon Signal zurück geben. Die Struktur befindet sich in Signals.h

Ein Kommentar wäre schön.. Zumindest möchte Korf das. Ich bin ja eigentlich fan davon, wenn der Code selbsterklärend ist, und das ist es hier finde ich, aber auf der anderen Seite ein kleines:

```cpp
/*
* @brief: returns first signal in signalBuffer from SignalGenerator
* if signalBuffer is empty it returns SIGNAL_BUFFER_EMPTY
*/
```

Reply:
wenn du das Signal noch in Signals.h dazufügen könntest wäre das cool!

Reply:
while((Signal signal = hal.getSignal).name != Signalname::SIGNAL_BUFFER_EMPTY)

Reply:
done